src/dune_rules/coq/coq_lib.ml

```diff
@@ -297,34 +293,44 @@ module DB = struct
 
     let rec find coq_db name =
       match coq_db.resolve name with
-      | `Theory (db, dir, stanza) -> Some (db, dir, stanza)
+      | `Theory (db, dir, stanza) -> `Theory (db, dir, stanza)
       | `Redirect coq_db -> find coq_db name
+      | `Stdlib lib -> `Stdlib lib
+      | `User_contrib lib -> `User_contrib lib
       | `Not_found -> (
         match coq_db.parent with
-        | None -> None
+        | None -> `Not_found
         | Some parent -> find parent name)
 
     let find coq_db ~coq_lang_version name =
       match find coq_db name with
-      | None -> `Not_found
-      | Some (mldb, dir, stanza) when coq_lang_version >= (0, 4) ->
-        `Found (mldb, dir, stanza)
-      | Some (mldb, dir, stanza) -> (
+      | `Not_found -> `Not_found
+      (* Composing with installed theories should come past 0.8 *)
```

Comment:
Why past 0.8 ? I think it should work out of the box once we build the lib db correctly.

Reply:
Same reason we restrict workspace composition to 0.4. Peoples Dune projects shouldn't suddenly start composing with installed theories.

Reply:
Yup, but we are bumping the lang no?

Reply:
Yes, but a new version of Dune should build a pre 0.8 project the same way.

Reply:
Ah sorry maybe I wasn't clear in the comment I mean >= 0.8

Reply:
We currently don't have that, we are waiting for 3.7 to land so we can bump 3.8 and then have 0.8.

Reply:
Oh indeed, I got confused by the "past" 0.8, sounds good then.